Error when two classes have the same name

```
[WARN] Issue detected with dagger.internal.codegen.ComponentProcessor. Expected 1 originating source file when generating /Users/lorenzo.quiroli/projects/caesium-android-v2/ClearScore/app/build/generated/source/kapt/debug/com/clearscore/mobile/core/provider/CSUserPreferences_Factory.java, but detected 0: [].
/Users/lorenzo.quiroli/projects/caesium-android-v2/ClearScore/app/build/tmp/kapt3/stubs/debug/com/clearscore/mobile/di/AppComponent.java:12: error: [Dagger/MapKeys] The same map key is bound more than once for java.util.Map<java.lang.Class<?>,javax.inject.Provider<dagger.android.AndroidInjector.Factory<?>>>
public abstract interface AppComponent {
                ^
      @Binds @IntoMap @ClassKey(com.clearscore.mobile.account.phonenumber.presentation.view.VerifyCodeFragment.class) @org.jetbrains.annotations.NotNull dagger.android.AndroidInjector.Factory<?> com.clearscore.mobile.account.di.AccountFragmentBuilder_BindPhoneVerifyCodeFragment.bindAndroidInjectorFactory(com.clearscore.mobile.account.di.AccountFragmentBuilder_BindPhoneVerifyCodeFragment.PhoneVerifyCodeFragmentSubcomponent.Factory)
      @Binds @IntoMap @ClassKey(com.clearscore.mobile.account.phonenumber.presentation.view.VerifyCodeFragment.class) @org.jetbrains.annotations.NotNull dagger.android.AndroidInjector.Factory<?> com.clearscore.mobile.account.di.AccountFragmentBuilder_BindVerifyCodeFragment.bindAndroidInjectorFactory(com.clearscore.mobile.account.di.AccountFragmentBuilder_BindVerifyCodeFragment.VerifyCodeFragmentSubcomponent.Factory)
      java.util.Map<java.lang.Class<?>,javax.inject.Provider<dagger.android.AndroidInjector.Factory<?>>> is injected at
          dagger.android.DispatchingAndroidInjector(injectorFactoriesWithClassKeys, …)
      dagger.android.DispatchingAndroidInjector<java.lang.Object> is provided at
          com.clearscore.mobile.di.user.UserComponent.injector() [com.clearscore.mobile.di.AppComponent → com.clearscore.mobile.di.user.UserComponent]/Users/lorenzo.quiroli/projects/caesium-android-v2/ClearScore/app/build/tmp/kapt3/stubs/debug/com/clearscore/mobile/di/AppComponent.java:12: error: [Dagger/MapKeys] The same map key is bound more than once for java.util.Map<java.lang.Class<?>,javax.inject.Provider<dagger.android.AndroidInjector.Factory<?>>>
public abstract interface AppComponent {
                ^
      @Binds @IntoMap @ClassKey(com.clearscore.mobile.account.phonenumber.presentation.view.VerifyNumberFragment.class) @org.jetbrains.annotations.NotNull dagger.android.AndroidInjector.Factory<?> com.clearscore.mobile.account.di.AccountFragmentBuilder_BindPhoneVerifyNumberFragment.bindAndroidInjectorFactory(com.clearscore.mobile.account.di.AccountFragmentBuilder_BindPhoneVerifyNumberFragment.PhoneVerifyNumberFragmentSubcomponent.Factory)
      @Binds @IntoMap @ClassKey(com.clearscore.mobile.account.phonenumber.presentation.view.VerifyNumberFragment.class) @org.jetbrains.annotations.NotNull dagger.android.AndroidInjector.Factory<?> com.clearscore.mobile.account.di.AccountFragmentBuilder_BindVerifyNumberFragment.bindAndroidInjectorFactory(com.clearscore.mobile.account.di.AccountFragmentBuilder_BindVerifyNumberFragment.VerifyNumberFragmentSubcomponent.Factory)
      java.util.Map<java.lang.Class<?>,javax.inject.Provider<dagger.android.AndroidInjector.Factory<?>>> is injected at
          dagger.android.DispatchingAndroidInjector(injectorFactoriesWithClassKeys, …)
      dagger.android.DispatchingAndroidInjector<java.lang.Object> is provided at
          com.clearscore.mobile.di.user.UserComponent.injector() [com.clearscore.mobile.di.AppComponent → com.clearscore.mobile.di.user.UserComponent]
```

--- a/compiler/src/main/java/dev/quiro/sheath/compiler/codegen/dagger/ContributesAndroidInjectorGenerator.kt
+++ b/compiler/src/main/java/dev/quiro/sheath/compiler/codegen/dagger/ContributesAndroidInjectorGenerator.kt
@@ -32,6 +32,7 @@ import dagger.multibindings.IntoMap
 import dev.quiro.sheath.compiler.codegen.PrivateCodeGenerator
 import dev.quiro.sheath.compiler.codegen.extractArrayOfClass
 import dev.quiro.sheath.compiler.codegen.extractSingleParameter
+import dev.quiro.sheath.compiler.codegen.requireFqName
 import dev.quiro.sheath.compiler.codegen.requireTypeReference
 import org.jetbrains.kotlin.descriptors.ModuleDescriptor
 import org.jetbrains.kotlin.name.FqName
@@ -73,7 +74,10 @@ internal class ContributesAndroidInjectorGenerator : PrivateCodeGenerator() {
     val returnType = function.requireTypeReference()
       .requireTypeName(module)
       .withJvmSuppressWildcardsIfNeeded(function)
-    val bindingTargetName = function.requireTypeReference().text
+    val bindingTargetName = function.requireTypeReference()
+      .requireFqName(module)
+      .shortName()
+      .asString()
     val className = "${clazz.generateClassName()}_${function.name!!.capitalize()}"
     val factoryClass = ClassName(packageName, className)
     val contributesAndroidInjector = function.findAnnotation(daggerContributesAndroidInjector)!!

--- a/compiler/src/test/java/dev/quiro/sheath/compiler/dagger/ContributesAndroidInjectorGeneratorTest.kt
+++ b/compiler/src/test/java/dev/quiro/sheath/compiler/dagger/ContributesAndroidInjectorGeneratorTest.kt
@@ -756,6 +756,73 @@ public abstract class DaggerModule1_BindMyFragment {
     }
   }
 
+  @Test
+  fun `two bind classes are generated for @ContributesAndroidInjector with named imports`() {
+    compile(
+      listOf(
+        """
+        package a.b.c
+        
+        class MyFragment
+        """,
+        """
+        package d.e.f
+        
+        class MyFragment
+        """,
+        """
+        package com.squareup.test
+        
+        import a.b.c.MyFragment
+        import d.e.f.MyFragment as OtherFragment
+        
+        @dagger.Module
+        abstract class DaggerModule1 {
+          @dagger.android.ContributesAndroidInjector
+          abstract fun bindMyFragment(): MyFragment
+          
+          @dagger.android.ContributesAndroidInjector
+          abstract fun bindOtherFragment(): OtherFragment
+        }
+        """
+      )
+    ) {
+      val subcomponentClass = daggerModule1.contributesAndroidInjector("BindOtherFragment")
+      assertThat(subcomponentClass.declaredConstructors).hasLength(1)
+
+      subcomponentClass.isAnnotationPresent(Module::class.java)
+      val moduleAnnotation = subcomponentClass.getAnnotation(Module::class.java)
+      assertThat(moduleAnnotation.subcomponents.single().java.name).isEqualTo(
+        "com.squareup.test.DaggerModule1_BindOtherFragment\$MyFragmentSubcomponent"
+      )
+
+      val bindMethod = subcomponentClass.declaredMethods.single()
+      assertThat(bindMethod.isAbstract).isTrue()
+      assertThat(bindMethod.isAnnotationPresent(Binds::class.java)).isTrue()
+      assertThat(bindMethod.isAnnotationPresent(IntoMap::class.java)).isTrue()
+      assertThat(bindMethod.isAnnotationPresent(ClassKey::class.java)).isTrue()
+      val classKeyAnnotation = bindMethod.getAnnotation(ClassKey::class.java)
+      assertThat(classKeyAnnotation.value.simpleName).isEqualTo("MyFragment")
+      assertThat(bindMethod.returnType).isEqualTo(AndroidInjector.Factory::class.java)
+      val parameter = bindMethod.parameters.single()
+      assertThat(parameter.type.name).isEqualTo(
+        "com.squareup.test.DaggerModule1_BindOtherFragment\$MyFragmentSubcomponent\$Factory"
+      )
+
+      val subcomponentInterface = classLoader.loadClass(
+        "com.squareup.test.DaggerModule1_BindOtherFragment\$MyFragmentSubcomponent"
+      )
+      val subcomponentFactoryInterface = classLoader.loadClass(
+        "com.squareup.test.DaggerModule1_BindOtherFragment\$MyFragmentSubcomponent\$Factory"
+      )
+      assertThat(subcomponentInterface.isAnnotationPresent(Subcomponent::class.java)).isTrue()
+      assertThat(subcomponentInterface.extends(AndroidInjector::class.java)).isTrue()
+      assertThat(subcomponentFactoryInterface.isAnnotationPresent(Subcomponent.Factory::class.java))
+        .isTrue()
+      assertThat(subcomponentFactoryInterface.extends(AndroidInjector.Factory::class.java)).isTrue()
+    }
+  }
+
   private fun compile(
     source: String,
     block: Result.() -> Unit = { }

